app/controllers/users_controller.rb

```diff
@@ -21,17 +21,23 @@ def show
     end
   end
 
+  # rubocop: disable Metrics/MethodLength
   def create
-    @user = User.new(user_params)
-    @user.provider = 'local'
-    if @user.save
-      @user.send_activation_email
-      flash[:info] = 'Please check your email to activate your account.'
-      redirect_to root_url
+    @user = User.find_by email: user_params[:email].downcase
+    if @user && !@user.activated
+      new_activation_digest
```

Comment:
Ruby method names should generally be verbs, so perhaps `regenerate_activation_digest`.